src/core/sync/condition.d

```diff
@@ -75,6 +75,20 @@ class Condition
      *  SyncError on error.
      */
     this( Mutex m ) nothrow @safe
+    {
+        this(m, true);
+    }
+
+    /// ditto
+    this( shared Mutex m ) shared nothrow @safe
+    {
+        this(m, true);
+    }
+
+    //
+    private this(this Q, M)( M m, bool _unused_ ) nothrow @safe
```

Comment:
This could also be a plain method instead as it does not initialize const/immutable data AFAICT.

Reply:
It may cause a breaking change in ABI level.
In the case of using a plain method, we have to use an unsafe cast in `this(shared Mutex)` and have to change the attribute of `this(shared Mutex)` from `@safe` to `@trusted` but it changes ABI of the constructor (See [Type Mangling](https://dlang.org/spec/abi.html#type_mangling)).

But I am not sure it really happens because I am not familiar with low level programming.

Reply:
@MoonlightSentinel Templating a member function on the type qualifier of `this` makes it possible to reuse code between `shared` and non-`shared` code without casts. This pattern is also used in `core.sync.mutex`.

Reply:
> It may cause a breaking change in ABI level.

Not really, it's a new symbol.

> In the case of using a plain method, we have to use an unsafe cast in `this(shared Mutex)` and have to change the attribute of `this(shared Mutex)` from `@safe` to `@trusted` [...]

You can always use a `@trusted` anonymous lamdba

Reply:
> @MoonlightSentinel Templating a member function on the type qualifier of `this` makes it possible to reuse code between `shared` and non-`shared` code without casts. This pattern is also used in `core.sync.mutex`.

Agreed, I was just thinking whether we need to have this kinda akward function header and  generate this constructor twice. 

But anyway, I'm fine if  this remains unchanged